### PR TITLE
INTMDB-420 cloud backup

### PR DESCRIPTION
**Pull request by rdharmedran**
## Description
 Feature update: cloud-provider-snapshots

Link to any related issue(s): 
https://jira.mongodb.org/browse/INTMDB-420
## Type of change:

- [✅]  Bug fix (non-breaking change which fixes an issue)
-[✅]  New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-[✅]  This change requires a documentation update

## Required Checklist:

-[✅]  I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments
 - Renamed the module cloud-provider-snapshots  into cloud-backup
 - Added input parameter validation
 - Read function returned already deleted snapshot.So we have added condition.
- Delete function was not checking if snapshot is already exist before delete it.We have added condition for the same.
- Added cfn-test configurations.

### CODE REVIEW

**Comment by Zuhairahmed** [comment]
@rdharmedran can we add GitHub checks to this PR (build + linting)?

File: cfn-resources/cloud-backup-snapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# MongoDB::Atlas::BackupSnapshot
+
+## Description
+Returns, adds,  and removes cloud backup.
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/README.md:4]:
```suggestion
Returns, takes, edits, and removes Cloud Backup snapshots
```
please follow Open API doc definitions https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups

File: cfn-resources/cloud-backup-snapshot/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Both CREATE & DELETE tests must pass.
+
+## Installation
+TAGS=logging make
+cfn submit --verbose --set-default
+
+## Cloudformation Examples
+
+Please use the [CFN Template](test/cloud-backup.sample-template.yaml)
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/README.md:38]: This link does not work, please double check 

```suggestion
Please see the [CFN Template](test/cloud-backup.sample-template.yaml)  for example resource
```

File: cfn-resources/cloud-backup-snapshot/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Integration Testing w/ AWS
+
+The [/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh](launch-x-quickstart.sh) script
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/README.md:42]: Link does not work, double check

File: cfn-resources/cloud-backup-snapshot/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
+# MongoDB::Atlas::BackupSnapshot
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/docs/README.md:1]: Fix resource name as mentioned above

**Review by Zuhairahmed — COMMENTED** [review]
Few doc comments below

File: cfn-resources/cloud-backup-snapshot/.rpdk-config [hunk 5]
```diff
@@ -1,12 +1,13 @@
 {
     "artifact_type": "RESOURCE",
-    "typeName": "MongoDB::Atlas::CloudProviderSnapshots",
+    "typeName": "MongoDB::Atlas::BackupSnapshot",
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/.rpdk-config:3]: Update name to "MongoDB::Atlas::CloudBackupSnapshot" here and throughout the resource

File: cfn-resources/cloud-backup-snapshot/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# MongoDB::Atlas::BackupSnapshot
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-backup-snapshot/README.md:1]: Update name to "MongoDB::Atlas::CloudBackupSnapshot" here and throughout the resource

File: examples/cloud-backup-snapshot/snapshot.json [hunk 7]
```diff
@@ -21,10 +21,10 @@
         }
       }
     },
-    "CloudProviderSnapshots": {
-      "Type": "MongoDB::Atlas::CloudProviderSnapshots",
+    "BackupSnapshot": {
+      "Type": "MongoDB::Atlas::BackupSnapshot",
```
Comment (Zuhairahmed) [inline examples/cloud-backup-snapshot/snapshot.json:25]: Update name to "MongoDB::Atlas::CloudBackupSnapshot" here and throughout the resource

**Review by Zuhairahmed — COMMENTED** [review]
Also fix name of resource to "MongoDB::Atlas::CloudBackupSnapshot"

File: cfn-resources/cloud-backup-snapshot/README.md (hunk 1 above)
Comment (rdharmedran) [reply cfn-resources/cloud-backup-snapshot/README.md]: This issue is fixed

File: cfn-resources/cloud-backup-snapshot/.rpdk-config (hunk 5 above)
Comment (rdharmedran) [reply cfn-resources/cloud-backup-snapshot/.rpdk-config]: This issue is fixed

File: cfn-resources/cloud-backup-snapshot/README.md (hunk 6 above)
Comment (rdharmedran) [reply cfn-resources/cloud-backup-snapshot/README.md]: This issue is fixed

**Review by swattyT — APPROVED** [review]
LGTM!

**Review by Zuhairahmed — APPROVED** [review]
LGTM! Thanks @rdharmedran for these updates

**Review by govardhanpagidi — APPROVED** [review]
LGTM

**Pull request merged**